### ISSUE TITLE
Docker client fails authorization after OVA reset

### ISSUE THREAD

**Issue by corrieb** [vmware/vic-product#891]
When I install the OVA, I go through the various processes to add the root certificate of the registry to a remote Docker and then I can login fine using my PSC vSphere ID. Eg.

```
vagrant@ubuntu-1604-vmware:~$ docker login 10.118.68.183
Username: bcorrie@vsphere.local
Password: 
Login Succeeded
```
I can do this without adding this ID anywhere in the Management UI. I don't need to be a member of a project or have any kind of role.

If I reset (power on / off) the OVA, I can no longer successfully login. I get the following message:

```
vagrant@ubuntu-1604-vmware:~$ docker login 10.118.68.183
Username: bcorrie@vsphere.local
Password: 
Error response from daemon: login attempt to https://10.118.68.183/v2/ failed with status: 401 Unauthorized
```
What's interesting is that the authentication appears to be working fine, but it fails authorization for some unknown reason. Even if I add this ID as an Admin to the default project, as a top level management Admin, to a new project. Doesn't matter. It still won't authorize me.

I tried re-downloading the root certificate thinking maybe it was regenerated. It was the same as the original one and didn't help.

The error I'm seeing in the Harbor logs is

```
Sep 27 17:36:16 172.18.0.1 registry[367]: time="2017-09-27T17:36:16.828684058Z" level=info msg="token signed by untrusted key with ID: \"6T3F:EYCE:II65:PYW7:6V3D:ZW57:H77J:J4RC:4A7U:ISI5:KST2:HEKE\""
Sep 27 17:36:16 172.18.0.1 registry[367]: time="2017-09-27T17:36:16.828813392Z" level=warning msg="error authorizing context: invalid token" go.version=go1.7.3 http.request.host=10.118.68.183 http.request.id=9ccce67b-22a7-46e2-b846-227c9784634c http.request.method=GET http.request.remoteaddr=10.118.69.174 http.request.uri="/v2/" http.request.useragent="docker/17.03.0-ce go/go1.7.5 git-commit/60ccb22 kernel/4.4.0-31-generic os/linux arch/amd64 UpstreamClient(Docker-Client/17.03.0-ce \\(linux\\))" instance.id=453a5f82-8977-4d66-94ed-f2c87f9aaaf3 service=registry version=v2.6.2
```
So what we need to figure out is:

1 - What is happening on boot that is causing this authorization problem?
2 - Does this mean that after a reboot, no user is able to pull or push images from the Registry? If so, then this is a ship stopper for 1.2.1 IMHO.

**Comment by corrieb** [comment]
Here are the OVA logs showing successful login followed by a reboot and unsuccessful login

[logs.zip](https://github.com/vmware/vic-product/files/1338385/logs.zip)

**Comment by andrewtchin** [comment]
@reasonerjt @sergiosagu Please take a look at this issue

**Comment by corrieb** [comment]
Note that this was using DHCP and I reset the OVA - hard power off / power on

**Comment by corrieb** [comment]
So it seems like this is less serious than first thought and we should document that you should always do a soft shutdown / restart of the OVA and list this as a symptom of a hard reset.

**Comment by lcastellano** [comment]
The problem appears to be between docker and Harbor, I checked Admiral and it is responding correctly with the Security Context (@sergiosagu you are off the hook for the moment). **When the Harbor container is restarted on the OVA, login starts working again**. Docker and Harbor use special Tokens to authenticate messages. I believe those are created and signed by Harbor, my theory is that Docker is sending the old token. I tried restarting the Docker daemon (I am not using a VCH for this test) but the problem persists. It would be interesting to see if the VCH has the same behavior.

**Comment by lcastellano** [comment]
I tried with the VCH and login was successful after reset of the OVA. So this appears to be a problem between Docker and Harbor.

**Comment by andrewtchin** [comment]
Thanks @lcastellano

**Comment by reasonerjt** [comment]
I did some investigation and I think I found the root cause.  
On the .ova deployment, there are two service `harbor.service` and `harbor_startup.service`.   `harbor_startup.service` will generate key pair for signing and verifying registry's token.  

After "Reset" the service `harbor.service` didn't wait for the finish of `harbor_startup.service`, so `registry` picked the old public key and `ui` picked the new private key, this mismatch caused the failure of `docker login`.


```
root@localhost [ /usr/lib/systemd/system ]# journalctl -u harbor_startup.service
-- Logs begin at Tue 2017-09-26 04:19:34 UTC, end at Thu 2017-09-28 04:00:03 UTC. --
Sep 26 04:23:03 localhost.localdomain systemd[1]: Starting Harbor Startup Configuration...
Sep 26 04:23:03 localhost.localdomain bash[1029]: Hostname is null, set it to IP
Sep 26 04:23:03 localhost.localdomain bash[1029]: Setting hostname
Sep 26 04:23:03 localhost.localdomain bash[1029]: Setting ui_url_protocol
Sep 26 04:23:03 localhost.localdomain bash[1029]: Setting ssl_cert
Sep 26 04:23:03 localhost.localdomain bash[1029]: Setting ssl_cert_key
Sep 26 04:23:03 localhost.localdomain bash[1029]: Setting secretkey_path
Sep 26 04:23:03 localhost.localdomain bash[1029]: unable to write 'random state'
Sep 26 04:23:03 localhost.localdomain bash[1029]: Setting managed key db_password
Sep 26 04:23:03 localhost.localdomain bash[1029]: unable to write 'random state'
Sep 26 04:23:03 localhost.localdomain bash[1029]: Setting managed key clair_db_password
Sep 26 04:23:04 localhost.localdomain bash[1029]: Setting admiral_url
Sep 26 04:23:04 localhost.localdomain bash[1029]: Enable harbor startup
Sep 26 04:23:04 localhost.localdomain bash[1029]: Created symlink from /etc/systemd/system/multi-user.target.wants/harbor_startup.service to /usr/lib/systemd/system/harbo
r_startup.service.
Sep 26 04:23:04 localhost.localdomain bash[1029]: Created symlink from /etc/systemd/system/multi-user.target.wants/harbor.service to /usr/lib/systemd/system/harbor.servic
e.
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated and saved secret to file: /data/harbor/secretkey
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/nginx/nginx.conf
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/adminserver/env
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/ui/env
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/registry/config.yml
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/db/env
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/jobservice/env
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/jobservice/app.conf
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/ui/app.conf
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated certificate, key file: /etc/vmware/harbor/common/config/ui/private_key.pem, cert file: /etc/vmware/harbor/co
mmon/config/registry/root.crt
Sep 26 04:23:07 localhost.localdomain python[1159]: Copying sql file for notary DB
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated certificate, key file: /etc/vmware/harbor/cert_tmp/notary-signer-ca.key, cert file: /etc/vmware/harbor/cert_
tmp/notary-signer-ca.crt
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated certificate, key file: /etc/vmware/harbor/cert_tmp/notary-signer.key, cert file: /etc/vmware/harbor/cert_tmp
/notary-signer.crt
Sep 26 04:23:07 localhost.localdomain python[1159]: Copying certs for notary signer
Sep 26 04:23:07 localhost.localdomain python[1159]: Copying notary signer configuration file
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/notary/server-config.json
Sep 26 04:23:07 localhost.localdomain python[1159]: Copying nginx configuration file for notary
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/nginx/conf.d/notary.server.conf
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated and saved secret to file: /data/harbor/defaultalias
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/notary/signer_env
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/clair/postgres_env
Sep 26 04:23:07 localhost.localdomain python[1159]: Generated configuration file: /etc/vmware/harbor/common/config/clair/config.yaml
Sep 26 04:23:07 localhost.localdomain python[1159]: The configuration files are ready, please use docker-compose to start the service.
Sep 26 04:23:07 localhost.localdomain systemd[1]: Started Harbor Startup Configuration.
-- Reboot --
Sep 28 03:28:37 localhost.localdomain systemd[1]: Starting Harbor Startup Configuration...
Sep 28 03:28:37 localhost.localdomain bash[2031]: Hostname is null, set it to IP
Sep 28 03:28:37 localhost.localdomain bash[2031]: Setting hostname
Sep 28 03:28:37 localhost.localdomain bash[2031]: Setting ui_url_protocol
Sep 28 03:28:37 localhost.localdomain bash[2031]: Setting ssl_cert
Sep 28 03:28:37 localhost.localdomain bash[2031]: Setting ssl_cert_key
Sep 28 03:28:37 localhost.localdomain bash[2031]: Setting secretkey_path
Sep 28 03:28:37 localhost.localdomain bash[2031]: unable to write 'random state'
Sep 28 03:28:38 localhost.localdomain bash[2031]: Skipping existing managed key db_password
Sep 28 03:28:38 localhost.localdomain bash[2031]: unable to write 'random state'
Sep 28 03:28:38 localhost.localdomain bash[2031]: Skipping existing managed key clair_db_password
Sep 28 03:28:38 localhost.localdomain bash[2031]: Setting admiral_url
Sep 28 03:28:38 localhost.localdomain bash[2031]: Enable harbor startup
Sep 28 03:28:42 localhost.localdomain python[2279]: loaded secret from file: /data/harbor/secretkey
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/nginx/nginx.conf
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/adminserver/env
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/ui/env
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/registry/config.yml
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/db/env
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/jobservice/env
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/jobservice/app.conf
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/ui/app.conf
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated certificate, key file: /etc/vmware/harbor/common/config/ui/private_key.pem, cert file: /etc/vmware/harbor/co
mmon/config/registry/root.crt
Sep 28 03:28:42 localhost.localdomain python[2279]: Copying sql file for notary DB
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated certificate, key file: /etc/vmware/harbor/cert_tmp/notary-signer-ca.key, cert file: /etc/vmware/harbor/cert_
tmp/notary-signer-ca.crt
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated certificate, key file: /etc/vmware/harbor/cert_tmp/notary-signer.key, cert file: /etc/vmware/harbor/cert_tmp
/notary-signer.crt
Sep 28 03:28:42 localhost.localdomain python[2279]: Copying certs for notary signer
Sep 28 03:28:42 localhost.localdomain python[2279]: Copying notary signer configuration file
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/notary/server-config.json
Sep 28 03:28:42 localhost.localdomain python[2279]: Copying nginx configuration file for notary
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/nginx/conf.d/notary.server.conf
Sep 28 03:28:42 localhost.localdomain python[2279]: loaded secret from file: /data/harbor/defaultalias
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/notary/signer_env
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/clair/postgres_env
Sep 28 03:28:42 localhost.localdomain python[2279]: Generated configuration file: /etc/vmware/harbor/common/config/clair/config.yaml
Sep 28 03:28:42 localhost.localdomain python[2279]: The configuration files are ready, please use docker-compose to start the service.
Sep 28 03:28:45 localhost.localdomain systemd[1]: Started Harbor Startup Configuration.
```



The log of the two services:
```
root@localhost [ /usr/lib/systemd/system ]# journalctl -u harbor.service
-- Logs begin at Tue 2017-09-26 04:19:34 UTC, end at Thu 2017-09-28 03:54:09 UTC. --
Sep 26 04:23:07 localhost.localdomain systemd[1]: Started Harbor Container Registry.
Sep 26 04:23:15 localhost.localdomain start_harbor.sh[1205]: Creating network "harbor_harbor" with the default driver
Sep 26 04:23:17 localhost.localdomain start_harbor.sh[1205]: Creating network "harbor_notary-mdb" with the default driver
Sep 26 04:23:20 localhost.localdomain start_harbor.sh[1205]: Creating network "harbor_notary-sig" with the default driver
Sep 26 04:23:23 localhost.localdomain start_harbor.sh[1205]: Creating network "harbor_harbor-clair" with the default driver
Sep 26 04:23:24 localhost.localdomain start_harbor.sh[1205]: Creating network "harbor_harbor-notary" with the default driver
Sep 26 04:23:25 localhost.localdomain start_harbor.sh[1205]: Creating harbor-log
Sep 26 04:23:26 localhost.localdomain start_harbor.sh[1205]: Creating harbor-db
Sep 26 04:23:26 localhost.localdomain start_harbor.sh[1205]: Creating harbor-adminserver
Sep 26 04:23:26 localhost.localdomain start_harbor.sh[1205]: Creating clair-db
Sep 26 04:23:26 localhost.localdomain start_harbor.sh[1205]: Creating notary-db
Sep 26 04:23:26 localhost.localdomain start_harbor.sh[1205]: Creating registry
Sep 26 04:23:44 localhost.localdomain start_harbor.sh[1205]: Creating harbor-ui
Sep 26 04:23:44 localhost.localdomain start_harbor.sh[1205]: Creating notary-signer
Sep 26 04:23:44 localhost.localdomain start_harbor.sh[1205]: Creating clair
Sep 26 04:24:16 localhost.localdomain start_harbor.sh[1205]: Creating notary-server
Sep 26 04:24:16 localhost.localdomain start_harbor.sh[1205]: Creating harbor-jobservice
Sep 26 04:24:16 localhost.localdomain start_harbor.sh[1205]: Creating nginx
Sep 26 04:24:35 localhost.localdomain start_harbor.sh[1205]: Attaching to harbor-log, harbor-adminserver, harbor-db, notary-db, registry, clair-db, notary-signer, clair,
harbor-ui, nginx, harbor-jobservice, notary-server
Sep 26 04:24:35 localhost.localdomain start_harbor.sh[1205]: harbor-adminserver | WARNING: no logs are available with the 'syslog' log driver
Sep 26 04:24:35 localhost.localdomain start_harbor.sh[1205]: harbor-db        | WARNING: no logs are available with the 'syslog' log driver
Sep 26 04:24:35 localhost.localdomain start_harbor.sh[1205]: notary-db        | WARNING: no logs are available with the 'syslog' log driver
Sep 26 04:24:35 localhost.localdomain start_harbor.sh[1205]: registry         | WARNING: no logs are available with the 'syslog' log driver
Sep 26 04:24:35 localhost.localdomain start_harbor.sh[1205]: clair-db         | WARNING: no logs are available with the 'syslog' log driver
Sep 26 04:24:35 localhost.localdomain start_harbor.sh[1205]: notary-signer    | WARNING: no logs are available with the 'syslog' log driver
Sep 26 04:24:35 localhost.localdomain start_harbor.sh[1205]: clair            | WARNING: no logs are available with the 'syslog' log driver
Sep 26 04:24:35 localhost.localdomain start_harbor.sh[1205]: harbor-ui        | WARNING: no logs are available with the 'syslog' log driver
Sep 26 04:24:35 localhost.localdomain start_harbor.sh[1205]: nginx            | WARNING: no logs are available with the 'syslog' log driver
Sep 26 04:24:35 localhost.localdomain start_harbor.sh[1205]: harbor-jobservice | WARNING: no logs are available with the 'syslog' log driver
Sep 26 04:24:35 localhost.localdomain start_harbor.sh[1205]: notary-server    | WARNING: no logs are available with the 'syslog' log driver
-- Reboot --
Sep 28 03:28:37 localhost.localdomain systemd[1]: Started Harbor Container Registry.
Sep 28 03:28:38 localhost.localdomain start_harbor.sh[2027]: Couldn't find env file: /etc/vmware/harbor/common/config/notary/signer_env
Sep 28 03:28:38 localhost.localdomain systemd[1]: harbor.service: Main process exited, code=exited, status=1/FAILURE
Sep 28 03:28:39 localhost.localdomain docker-compose[2309]: Couldn't find env file: /etc/vmware/harbor/common/config/notary/signer_env
Sep 28 03:28:39 localhost.localdomain systemd[1]: harbor.service: Control process exited, code=exited status=1
Sep 28 03:28:40 localhost.localdomain docker-compose[2441]: Couldn't find env file: /etc/vmware/harbor/common/config/notary/signer_env
Sep 28 03:28:40 localhost.localdomain systemd[1]: harbor.service: Control process exited, code=exited status=1
Sep 28 03:28:40 localhost.localdomain systemd[1]: harbor.service: Unit entered failed state.
Sep 28 03:28:40 localhost.localdomain systemd[1]: harbor.service: Failed with result 'exit-code'.
Sep 28 03:28:45 localhost.localdomain systemd[1]: harbor.service: Service hold-off time over, scheduling restart.
Sep 28 03:28:45 localhost.localdomain systemd[1]: Stopped Harbor Container Registry.
Sep 28 03:28:45 localhost.localdomain systemd[1]: Started Harbor Container Registry.
Sep 28 03:28:46 localhost.localdomain start_harbor.sh[3277]: harbor-log is up-to-date
Sep 28 03:28:46 localhost.localdomain start_harbor.sh[3277]: Recreating harbor-adminserver
Sep 28 03:28:46 localhost.localdomain start_harbor.sh[3277]: Starting harbor-db
Sep 28 03:28:46 localhost.localdomain start_harbor.sh[3277]: clair-db is up-to-date
Sep 28 03:28:46 localhost.localdomain start_harbor.sh[3277]: registry is up-to-date
Sep 28 03:28:46 localhost.localdomain start_harbor.sh[3277]: notary-db is up-to-date
Sep 28 03:28:46 localhost.localdomain start_harbor.sh[3277]: clair is up-to-date
Sep 28 03:28:46 localhost.localdomain start_harbor.sh[3277]: notary-signer is up-to-date
Sep 28 03:28:46 localhost.localdomain start_harbor.sh[3277]: notary-server is up-to-date
Sep 28 03:28:48 localhost.localdomain start_harbor.sh[3277]: Recreating harbor-ui
Sep 28 03:28:50 localhost.localdomain start_harbor.sh[3277]: Recreating harbor-jobservice
Sep 28 03:28:50 localhost.localdomain start_harbor.sh[3277]: Recreating nginx
Sep 28 03:28:54 localhost.localdomain start_harbor.sh[3277]: Attaching to harbor-log, clair-db, registry, notary-db, clair, notary-signer, notary-server, harbor-db, harbo
r-adminserver, harbor-ui, harbor-jobservice, nginx
Sep 28 03:28:54 localhost.localdomain start_harbor.sh[3277]: clair-db         | WARNING: no logs are available with the 'syslog' log driver
Sep 28 03:28:54 localhost.localdomain start_harbor.sh[3277]: registry         | WARNING: no logs are available with the 'syslog' log driver
Sep 28 03:28:54 localhost.localdomain start_harbor.sh[3277]: notary-db        | WARNING: no logs are available with the 'syslog' log driver
Sep 28 03:28:54 localhost.localdomain start_harbor.sh[3277]: clair            | WARNING: no logs are available with the 'syslog' log driver
Sep 28 03:28:54 localhost.localdomain start_harbor.sh[3277]: notary-signer    | WARNING: no logs are available with the 'syslog' log driver
Sep 28 03:28:54 localhost.localdomain start_harbor.sh[3277]: notary-server    | WARNING: no logs are available with the 'syslog' log driver
Sep 28 03:28:54 localhost.localdomain start_harbor.sh[3277]: harbor-db        | WARNING: no logs are available with the 'syslog' log driver
Sep 28 03:28:54 localhost.localdomain start_harbor.sh[3277]: harbor-adminserver | WARNING: no logs are available with the 'syslog' log driver
Sep 28 03:28:54 localhost.localdomain start_harbor.sh[3277]: harbor-ui        | WARNING: no logs are available with the 'syslog' log driver
Sep 28 03:28:54 localhost.localdomain start_harbor.sh[3277]: harbor-jobservice | WARNING: no logs are available with the 'syslog' log driver
Sep 28 03:28:54 localhost.localdomain start_harbor.sh[3277]: nginx            | WARNING: no logs are available with the 'syslog' log driver
```

Note after the `-- Reboot --` the `harbor.service` is triggered at `Sep 28 03:28:37` which is before the end of `harbor_startup.service`, which was at `Sep 28 03:28:45`

**Comment by reasonerjt** [comment]
@andrewtchin @corrieb  To fix this issue the sequence of triggering the services has to be consistent after "reset"

**Comment by stuclem** [comment]
Adding the release notes flag. If this is likely to be fixed in the near future, it belongs in the release notes. If you consider it necessary, I can also add a troubleshooting topic in the core docs.

**Comment by corrieb** [comment]
@stuclem It would be easy to accidentally do a hard reset instead of a reboot and then be perplexed as to why it results in these symptoms. For that reason, we should definitely document in a way that GSS can respond to.

**Comment by lcastellano** [comment]
To continue on Daniel's analysis, both services seem to be "enabled" in systemd (from configure-harbor.sh)

# Start on startup
echo "Enable harbor startup"
systemctl enable harbor_startup.service
systemctl enable harbor.service

That would mean that the two services are started automatically on reboot. Which implies that they will run concurrently (there are no dependencies between them). Which is what might be causing the problem in Harbor upon reset. If this happened all the times, there should not be any difference between reboot and reset. But there is, as a soft reboot does not cause the problem with docker login.  The only difference I can think of is that systemd executes the commands in ExecStop and ExecStopPost in the harbor.service, in case of soft shutdown, while it does not in case of hard reset. I think we will need to look into our systemd service dependencies in the OVA for next release as it appears to have some inconsistencies. I would not touch it for the current release 1.2.1 as it will require a lot of testing.

**Comment by stuclem** [comment]
Here's a draft release note entry:

-------


- <b>Attempts to log in to vSphere Integrated Containers Registry from Docker fail with error 401 Unauthorized [#891](https://github.com/vmware/vic-product/issues/891)</b><br>When you attempt to log in to vSphere Integrated Containers Registry, you see a 401 Unauthorized error, even though you have correctly configured the Docker client with the registry certificate and you have successfully logged in to that registry instance before. This issue can occur if the  vSphere Integrated Containers appliance has experienced a hard reboot. 

   Workaround: In the vSphere Client, restart the vSphere Integrated Containers appliance by selecting **Power** > **Restart Guest OS**. 

--------

@corrieb @lcastellano does this cover it? I'll also add a troubleshooting topic to the core doc.

**Comment by stuclem** [comment]
If this is going to be fixed as a high priority in 1.3, then I think that release notes in 1.2 suffice. Removing the user-doc flag.

**Comment by stuclem** [comment]
RN reviewed by @corrieb via Slack. Removing note flag.
